src/libraries/System.Private.CoreLib/src/System/Number.BigInteger.cs

```diff
@@ -1042,14 +1020,24 @@ public bool IsZero()
 
             public void Multiply(uint value)
             {
-                Multiply(ref this, value, ref this);
+                Multiply(ref this, value, out this);
             }
 
             public void Multiply(ref BigInteger value)
             {
-                BigInteger temp = new BigInteger(0);
-                temp.SetValue(ref this);
-                Multiply(ref temp, ref value, ref this);
+                switch (value._length)
```

Comment:
Is the codegen for the switch case here good? I would expect that an if statement might have better codegen and be better predicted by the CPU.

Reply:
I can't compile it to test on my machine at the moment, but I would have figured switch would be fine; I don't know much about how prediction works with it.
I'm reconsidering this change because it doesn't gain us much. It will still be O(nm) regardless of the checks, the only difference being avoiding a copy. The checks inside `Multiply(ref BigInteger, ref BigInteger, out BigInteger)` should be sufficient for choosing the best way to complete the operation.